docs/src/developer/manual/coding-conventions.md

```diff
@@ -1,2 +1,558 @@
+  unless it is an identifier that begins with a lower case letter (never alter
+  the case of identifiers!).
+
+We use the following style for documenting functions and class methods:
+
+```python
+
+class MyObject(object):
+
+   """My new class."""
+
+    def set_keyword_db_path(self, path):
+        """Set the path for the keyword database (sqlite).
+
+        The file will be used to search for keywords for non local datasets.
+
+        :param path: A valid path to a sqlite database. The database does
+            not need to exist already, but the user should be able to write
+            to the path provided.
+        :type path: str
+
+        :returns: Flag indicating if the path was set successfully.
+        :rtype: boolean
+        """
+        self.keyword_db_path = str(path)
+```
+
+Another example:
+
+```
+class MyObject(object):
+
+   """My new class."""
+
+    def add_layers(scenario_dir, paths):
+        """Add the layers described in a scenario file to QGIS.
+
+        :param scenario_dir: Base directory to find path.
+        :type scenario_dir: str
+
+        :param paths: Path of scenario file (or a list of paths).
+        :type paths: str, list
+
+        :raises: Exception, TypeError, FileNotFoundError
+
+        .. note::
+            * Exception - occurs when paths have illegal extension
+            * TypeError - occurs when paths is not string or list
+            * FileNotFoundError - occurs when file not found
+    """
+```
+
+Note the following in the above examples:
+
+* The first line of a docstring should be a precis of the class/method/function
+  expressed in less than 80 chars, terminated with a full stop and exclude
+  redundant phrases such as 'Class to do x' or 'This method does...'.
+* There should be an empty line following the first docstring line.
+* More detailed explanation and usage examples can follow this first line. The
+  detailed explanation should not repeat the information provided in the
+  parameters and returns sections.
+* A line break should follow the optional detailed description.
+* **param** and **type** are grouped together with no line break between them.
+* If the param description is more than one line, indent the successive lines
+  with 4 spaces.
+* A newline should be placed after each type and rtype.
+* If multiple types are allowed, separate them with commas e.g. ``:rtype: str,
+  boolean``.
+* If a function or method returns nothing, no **returns** section is used.
+* If a function or method does not raise anything explicitly, no raises section
+  is used.
+* If a function or method is extremely obvious there is no need to have
+  anything more than a single line docstring.
+* If a function or method returns a tuple it should be be documented as
+  ``:rtype: (<type>, <type>, ..)`` e.g. ``:rtype: (int, int)``.
+
+Please also see the **api documentation how-to** section for more
+information on how to document your code properly.
```

Comment:
Which document/section of documentation does this refer to?

We should be able to make this a functional link